src/Conversion/ONNXToKrnl/Tensor/Pad.cpp

```diff
@@ -68,9 +68,23 @@ struct ONNXPadOpLowering : public OpConversionPattern<ONNXPadOp> {
       // This way is to avoid using `select` in computing indices as doing for
       // 'edge' and 'reflect' modes.
       Value cValue;
+      bool useConstantIndexZero = false;
+      if (auto constantValueType =
+              constantValue.getType().dyn_cast<MemRefType>()) {
+        if (constantValueType.getElementType().isF32() &&
+            constantValueType.getRank() == 1) {
+          // If the constant_value type is 1xf32 do krnl.load with an index of
+          // 0 to avoid an assertion failure in AffineLoadOp::build().
+          useConstantIndexZero = true;
+        }
+      }
       if (constantValue.getType().isa<NoneType>()) {
         // Default to 0 if constant_value is not specified.
         cValue = create.math.constant(resElementType, 0);
+      } else if (useConstantIndexZero == true) {
+        // Do krnl.load with an index of 0 when constant_value type is 1xf32.
+        auto idx = create.math.constantIndex(0);
```

Comment:
Nit: only use `auto` for lambda functions or really ugly types. Please replace with `Value`

Reply:
Update to not use auto.  Thanks!